File: go/14_json.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type Person struct {
	Name  string   `json:"name"`
	Age   int      `json:"age"`
	Langs []string `json:"langs"`
}

func main() {
	p := Person{Name: "ulas", Age: 20, Langs: []string{"go", "rust"}}

	b, _ := json.MarshalIndent(p, "", "  ")
	fmt.Println(string(b))

	var parsed Person
	json.Unmarshal(b, &parsed)
	fmt.Println(parsed.Name)

	os.WriteFile("data.json", b, 0644)
}

File: rust/14_json.rs
// 14 json - needs serde in real projects
// shown conceptually

fn main() {
    // In real Rust you use serde + serde_json
    // cargo add serde serde_json --features serde/derive

    let json = r#"{"name": "ulas", "age": 20}"#;
    println!("raw json: {}", json);
    println!("Use serde_json::from_str to parse");
}
